docs/deployment-ux.md

```diff
@@ -300,111 +310,193 @@ impacts
-    ``` {.shell}
-    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
-    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
-    SomeEnvironment/Prod:1    Undeploying  0         4         4            4        
-    SomeEnvironment/Prod:2    Deploying    5         1         1            1        
+    ```
+    $ aws ecs describe-environment-status --all --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT
+    SomeEnvironment/Prod:1    Undeploying  0         4
+    SomeEnvironment/Prod:2    Deploying    5         1
     ```
 
     Deployment progresses in a similar fashion until the last task of
     the old revision is left:
 
-    ``` {.shell}
-    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
-    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
-    SomeEnvironment/Prod:1    Undeploying  0         1         1            1        
-    SomeEnvironment/Prod:2    Deploying    5         5         5            5        
+    ```
+    $ aws ecs describe-environment-status --all --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT
+    SomeEnvironment/Prod:1    Undeploying  0         1
+    SomeEnvironment/Prod:2    Deploying    5         5
     ```
 
     Deployment completes once the final task in the old revision is
     terminated:
 
-    ``` {.shell}
-    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
-    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
-    SomeEnvironment/Prod:1    Inactive     0         0         0            0        
-    SomeEnvironment/Prod:2    Active       5         5         5            5        
     ```
+    $ aws ecs describe-environment-status --all --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT
+    SomeEnvironment/Prod:1    Inactive     0         0
+    SomeEnvironment/Prod:2    Active       5         5
+    ```
+
+#### Changes to cluster membership while a deployment is ongoing
+
+For larger clusters, or clusters that are backed by Autoscaling Groups, the available instances in the cluster might change (sometimes dramatically). Since the total number of available instances determines the desired end-state of the Environment, these state changes could affect the ability of an Environment to make progress while honoring its deployment constraints.
+
+The behaviour we're aiming for is to minimize the number of instances that are not running the Daemon task at all at any given point in time. If a configurable number of instances are without the correct version of a Daemon task for a configurable amount of time, the deployment will enter the `Failing` state and emit an alarm. However, it will continue to try and make progress.
```

Comment:
It's not clear why the instances won't have the correct version and result in a failing state.

Reply:
Ah, I think I meant "EnvironmentRevision" here, not "version". Basically, if the scheduler has to choose whether to

1. first replace an existing, older revision of the environment's task on an instance, or
2. start the current revision of the environment's task on a host that doesn't have any task running on it at all,

 it should prioritize the latter.

I'll see if I can reword this to be clearer.